src/python_testing/TC_EEVSE_2_6.py

```diff
@@ -0,0 +1,293 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs: run1
+# test-runner-run/run1/app: ${ENERGY_MANAGEMENT_APP}
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/quiet: True
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json --enable-key 000102030405060708090a0b0c0d0e0f --application evse
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --hex-arg enableKey:000102030405060708090a0b0c0d0e0f --endpoint 1 --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+# === END CI TEST ARGUMENTS ===
+
+import logging
+import time
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from matter_testing_support import (ClusterAttributeChangeAccumulator, EventChangeCallback, MatterBaseTest, TestStep,
+                                    async_test_body, default_matter_test_main)
+from mobly import asserts
+from TC_EEVSE_Utils import EEVSEBaseTestHelper
+
+logger = logging.getLogger(__name__)
+
+
+class TC_EEVSE_2_6(MatterBaseTest, EEVSEBaseTestHelper):
+
+    def desc_TC_EEVSE_2_6(self) -> str:
+        """Returns a description of this test"""
+        return "5.1.6. [TC-EEVSE-2.6] Test Q quality functionality with DUT as Server"
+
+    def pics_TC_EEVSE_2_6(self):
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+
+        return ["EEVSE.S"]
+
+    def steps_TC_EEVSE_2_6(self) -> list[TestStep]:
+        steps = [
+            TestStep("1", "Commission DUT to TH (can be skipped if done in a preceding test)"),
+            TestStep("2", "TH reads from the DUT the FeatureMap",
```

Comment:
```suggestion
            TestStep("2", "The TH reads the FeatureMap from the DUT",
```

Reply:
This wording comes from the test plan verbatim, which uses a macro for {THRead}. That expands to this text.

I agree it sounds clunky in English, but that’s how the test plans are written!